hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/util/RecoverLeaseFSUtils.java

```diff
@@ -167,21 +192,25 @@ private static boolean checkIfTimedout(final Configuration conf, final long reco
    * Try to recover the lease.
    * @return True if dfs#recoverLease came by true.
    */
-  private static boolean recoverLease(final DistributedFileSystem dfs, final int nbAttempt,
-    final Path p, final long startWaiting) throws FileNotFoundException {
+  private static boolean recoverLease(final Object dfs, final int nbAttempt, final Path p,
+    final long startWaiting) throws FileNotFoundException {
     boolean recovered = false;
     try {
-      recovered = dfs.recoverLease(p);
+      recovered = (Boolean) dfs.getClass().getMethod("recoverLease", new Class[] { Path.class })
```

Comment:
We usually determine the handle in a staic initializer, and use the stored handle in the calls.

Reply:
Maybe move the reflection logic to FSUtils ?